src/pages/SelectLanguage/SelectLanguage.js

```diff
@@ -0,0 +1,96 @@
+import { useState, useEffect, React } from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+import { Picker } from '@react-native-picker/picker'
+import { useSelector, useDispatch } from 'react-redux'
+
+import colors from '../../theme/colors'
+import languages from './languageList'
+import {
+  setInputLanguage,
+  setOutputLanguage,
+} from '../../slices/targetLanguageSlice'
+
+export default function SelectLanguage() {
+  const currInput = useSelector((state) => state.targetLanguage.input)
+  const currOutput = useSelector((state) => state.targetLanguage.output)
+  const [chosenInput, setChosenInput] = useState(currInput)
+  const [chosenOutput, setChosenOutput] = useState(currOutput)
+
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(setInputLanguage(chosenInput))
+  }, [chosenInput])
+
+  useEffect(() => {
+    dispatch(setOutputLanguage(chosenOutput))
+  }, [chosenOutput])
+
+  const onInputValueChange = (itemValue) => {
+    setChosenInput(itemValue)
+    dispatch(setInputLanguage(chosenInput))
```

Comment:
what will be the value of "itemValue"? is that the "value" field like on line 54? In that case won't it update the visible value to the user to the LanguageCode (E.g.. en-US) instead of the LanguageName?

If that's the case... to avoid confusion I would follow how I described last night... just store the LanguageNames in your state, then we'll pass those Name strings to the server and translate them to codes on that side. Meaning create an array with those LanguageNames (e.g. Languages = ['English (US)', 'French', 'German', 'Hindi',....]) and map over that.